compiler/int128.nim

```diff
@@ -540,7 +540,14 @@ proc ldexp(x: float64, exp: cint): float64 {.importc: "ldexp", header: "<math.h>
 
 template bitor(a, b, c: Int128): Int128 = bitor(bitor(a, b), c)
 
+proc inInt128Range*(arg: float64): bool =
+  const lowerBound = -pow(2'f64, 127)
+  const upperBound = +pow(2'f64, 127) # 2^127-1 can't be represented with float64 precision
+  lowerBound <= arg and arg < upperBound
+
 proc toInt128*(arg: float64): Int128 =
+  # actually enabling this assert will cause /tests/misc/tconv.nim to fail
+  # assert(inInt128Range(arg), "out of range")
```

Comment:
WDYT about producing a saturating conversion as a fallback? ie. if the value is too large then construct the largest int128, and similarly the smallest int128 for values that are too small. That way it'd make sure that every callsite is protected from the undefined behavior of this conversion.

Reply:
that is actually a good idea.

Reply:
done